graph2tac/tfgnn/train.py

```diff
@@ -306,9 +306,10 @@ def run(self,
                                  metrics=self.prediction_task.metrics())
 
         # get training data
-        train_proofstates, valid_proofstates = self.dataset.proofstates(shuffle=True)
+        train_proofstates = self.dataset.proofstates(TRAIN, True)
+        valid_proofstates = self.dataset.proofstates(VALID, False)
         if self.definition_task:
-            definitions = self.dataset.definitions(TRAIN, shuffle=False)
+            definitions = self.dataset.definitions(TRAIN, False)
```

Comment:
This is actually a bug not from this PR but from the previous one.  I discovered it this week and it leads to a fairly large regression in our model.  Can you make this `shuffle=True` again?

Reply:
Definitions need to be shuffled like all other training data.  (Also, there is a good reason to shuffle definitions for validation as well.  Since there are more definitions than validation proof states (I think), to get a good validation statistic for definition loss, we should have a random subset of definitions.  Of course, one day we may just want separate definitions for validation and training.)

Reply:
done